pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -355,25 +350,23 @@ func (r *ReconcileIBMBlockCSI) updateStatus(instance *ibmblockcsi.IBMBlockCSI, o
 				logger.Error(err, "failed to get controller pod")
 				return err
 			}
-			if originalStatus.ControllerReady || !r.areAllPodImagesSynced(controllerStatefulset, controllerPod) {
-				logger.Info("controller requires restart",
-							"ReadyReplicas", controllerStatefulset.Status.ReadyReplicas,
-							"Replicas", controllerStatefulset.Status.Replicas)
-				controllerRestart = true
+	
+			for _, containerStatus := range controllerPod.Status.ContainerStatuses {
+				if containerStatus.State.Waiting != nil {
+					if containerStatus.State.Waiting.Reason == "ImagePullBackOff" && !r.areAllPodImagesSynced(controllerStatefulset, controllerPod) {
+						logger.Info("controller requires restart",
+												"ReadyReplicas", controllerStatefulset.Status.ReadyReplicas,
+												"Replicas", controllerStatefulset.Status.Replicas)
+						r.restartControllerPod(controllerPod)
+					}
+				}
 			}
 		}
-
-
-		if originalStatus.NodeReady && !instance.Status.NodeReady {
-			logger.Info("node rollout requires restart",
-				"DesiredNumberScheduled", node.Status.DesiredNumberScheduled,
-				"NumberAvailable", node.Status.NumberAvailable)
-			nodeRolloutRestart = true
-		}
 		phase = csiv1.DriverPhaseCreating
 	}
 	instance.Status.Phase = phase
 	instance.Status.Version = oversion.DriverVersion
+	logger.Info("updating IBMBlockCSI status")
```

Comment:
I don't think this is true (it's true only inside the following `if`)

Reply:
done